File: src/main/java/com/crm/service/CustomerService.java
package com.crm.service;

import com.crm.dto.request.CustomerCreateRequest;
import com.crm.dto.request.CustomerUpdateRequest;
import com.crm.dto.response.CustomerResponse;
import com.crm.util.PagedResponse;
import org.springframework.data.domain.Pageable;

public interface CustomerService {
    CustomerResponse create(CustomerCreateRequest request);

    CustomerResponse update(Long id, CustomerUpdateRequest request);

    CustomerResponse getById(Long id);

    PagedResponse<CustomerResponse> list(Pageable pageable);

    void delete(Long id);
}

File: crm-frontend/src/services/customerService.js
import httpClient from '../api/httpClient'

export const customerService = {
  async getAll(page = 0, size = 10) {
    const response = await httpClient.get('/customers', {
      params: { page, size },
    })
    return response.data
  },

  async getById(id) {
    const response = await httpClient.get(`/customers/${id}`)
    return response.data
  },

  async create(customerData) {
    const response = await httpClient.post('/customers', customerData)
    return response.data
  },

  async update(id, customerData) {
    const response = await httpClient.put(`/customers/${id}`, customerData)
    return response.data
  },

  async delete(id) {
    const response = await httpClient.delete(`/customers/${id}`)
    return response.data
  },
}
